codec-http/src/main/java/io/netty/handler/codec/http/HttpHeaderDateFormatter.java

```diff
@@ -0,0 +1,425 @@
+        timeFound = false;
+        hours = -1;
+        minutes = -1;
+        seconds = -1;
+        dayOfMonthFound = false;
+        dayOfMonth = -1;
+        monthFound = false;
+        month = -1;
+        yearFound = false;
+        year = -1;
+        cal.set(Calendar.MILLISECOND, 0);
+        sb.setLength(0);
+    }
+
+    private boolean tryParseTime(CharSequence txt, int tokenStart, int tokenEnd) {
+        int len = tokenEnd - tokenStart;
+
+        // h:m:s to hh:mm:ss
+        if (len < 5 || len > 8) {
+            return false;
+        }
+
+        int localHours = -1;
+        int localMinutes = -1;
+        int localSeconds = -1;
+        int currentPartNumber = 0;
+        int currentPartValue = 0;
+
+        for (int i = tokenStart; i < tokenEnd; i++) {
+            char c = txt.charAt(i);
+            if (isDigit(c)) {
+                currentPartValue = currentPartValue * 10 + getNumericalValue(c);
+            } else if (c == ':') {
+                if (currentPartValue == 0) {
+                    // invalid :: (nothing in between)
+                    return false;
+                }
+                switch (currentPartNumber) {
+                    case 0:
+                        // flushing hours
+                        localHours = currentPartValue;
+                        currentPartValue = 0;
+                        currentPartNumber++;
+                        break;
+                    case 1:
+                        // flushing minutes
+                        localMinutes = currentPartValue;
+                        currentPartValue = 0;
+                        currentPartNumber++;
+                        break;
+                    default:
+                        // invalid, too many :
+                        return false;
+                }
+            } else {
+                // invalid char
+                return false;
+            }
+        }
+
+        if (currentPartValue > 0) {
+            // pending seconds
+            localSeconds = currentPartValue;
+        }
+
+        if (localHours >= 0 && localMinutes >= 0 && localSeconds >= 0) {
+            hours = localHours;
+            minutes = localMinutes;
+            seconds = localSeconds;
+            return true;
+        }
+
+        return false;
+    }
+
+    private boolean tryParseDayOfMonth(CharSequence txt, int tokenStart, int tokenEnd) {
+        int len = tokenEnd - tokenStart;
+
+        if (len == 1) {
+            char c0 = txt.charAt(tokenStart);
+            if (isDigit(c0)) {
+                dayOfMonth = getNumericalValue(c0);
+                return true;
+            }
+
+        } else if (len == 2) {
+            char c0 = txt.charAt(tokenStart);
+            char c1 = txt.charAt(tokenStart + 1);
+            if (isDigit(c0) && isDigit(c1)) {
+                dayOfMonth = getNumericalValue(c0) * 10 + getNumericalValue(c1);
+                return true;
+            }
+        }
+
+        return false;
```

Comment:
I'm sorry, but I really fails to see how (without using different variables).
The example below definitively doesn't compile:
```java
switch(tokenEnd - tokenStart) {
    case 1:
        char c0 = txt.charAt(tokenStart);
        if (isDigit(c0)) {
            dayOfMonth = getNumericalValue(c0);
            return true;
        }
        return false;
    case 2:
        char c0 = txt.charAt(tokenStart);
        char c1 = txt.charAt(tokenStart + 1);
        if (isDigit(c0) && isDigit(c1)) {
            dayOfMonth = getNumericalValue(c0) * 10 + getNumericalValue(c1);
            return true;
        }
        return false;
    default:
        return false;
}
```

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.5.1:compile (default-compile) on project netty-codec-http: Compilation failure
[ERROR] /Users/slandelle/Documents/dev/IdeaProjects/netty-project/netty/codec-http/src/main/java/io/netty/handler/codec/http/HttpHeaderDateFormatter.java:[240,13] error: variable c0 is already defined in method tryParseDayOfMonth(CharSequence,int,int)
```

Am I missing something?